FeedAPIChallenge/Feed API/FeedImageResponseMapper.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  FeedImageResponseMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Антон on 19.08.21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class FeedImageResponseMapper {
+	static func mapFeedImage(data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
+		guard response.statusCode == 200 else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+		do {
+			let itemsContainer = try JSONDecoder().decode(FeedImageResponse.self, from: data)
+			return .success(itemsContainer.feed)
+		} catch {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
```

Comment:
Since error from `catch` block is ignored the `do ... catch` can be simplified to optional `try?`
```swift
guard response.statusCode == 200 else {
	return .failure(RemoteFeedLoader.Error.invalidData)
}
guard let itemsContainer = try? JSONDecoder().decode(FeedImageResponse.self, from: data) else {
	return .failure(RemoteFeedLoader.Error.invalidData)
}
return .success(itemsContainer.feed)
```

And then two guards can be combined into one so that the code is shorter:
```suggestion
		guard response.statusCode == 200, let itemsContainer = try? JSONDecoder().decode(FeedImageResponse.self, from: data) else {
			return .failure(RemoteFeedLoader.Error.invalidData)
		}
		return .success(itemsContainer.feed)
```